lib/mapping/CMapHeader.h

```diff
@@ -244,10 +245,16 @@ class DLL_LINKAGE CMapHeader
 
 	/// "main quests" of the map that describe victory and loss conditions
 	std::vector<TriggeredEvent> triggeredEvents;
+	
+	/// translations for map to be transferred over network
+	JsonNode translations;
```

Comment:
Why not store this in a form that is more clear and defined in compile time? e.g.
```cpp
std::map<std::string, std::map<std::string, std::string>>
```

Reply:
Ideally, translations should be defined our of the class, e.g MapTranslations (but literally it'd be same JsonNode with some syntax sugar). But currently all the translations logic is encapsulated (including serialisation), so it's not supposed to work with this field after loading and that's why no parsing here.